### ISSUE TITLE
[C++] Reference relations are not generated for constructors using list initialization

### ISSUE THREAD

**Issue by gsrohde** [doxygen/doxygen#8102]
**Describe the bug**

If a constructor that calls a function uses the _list_ initialization syntax to initialize a data member, the _references/referenced-by_ links are not generated for that call.

**Expected behavior**

When the configuration file sets REFERENCED_BY_RELATION and REFERENCES_RELATION to YES, links to and from a function called by a constructor should be generated.

**To Reproduce**

Make a Doxyfile with this content:
```
# Doxyfile 1.8.20

EXTRACT_ALL            = YES
INPUT                  = AB.cpp
REFERENCED_BY_RELATION = YES
REFERENCES_RELATION    = YES
```
and a source file `AB.cpp` containing
```
void helper() {}

class A
{
public:
    A(int a) : aa{a} {
        helper();
    }
private:
    int aa;
};

class B {
public:
    B(int b) : bb(b) {
        helper();
    }
private:
    int bb;
};
```
Generate the Doxygen documentation and view `html/class_a.html`.  The documentation for the constructor `A()` should say "references `helper()`" but doesn't.  Then view `html/class_b.html`.  The documentation for the constructor for class B, which uses the "()" initialization syntax instead of the "{}" syntax, _does_ include the reference link.  Then click this link to go to the documentation for `helper()`.  This documentation states that `helper()` is "Referenced by B::B()."  But no mention is made of `A::A()`.

**Version**

I used version 1.8.20 on a Mac that was installed from the DMG file.

**Comment by albert-github** [comment]
I think this is called aggregate-initialization and as far as I know not (yet) supported by doxygen.

**Comment by doxygen** [comment]
@gsrohde Please verify if the commit I just pushed solves the problem.

**Comment by gsrohde** [comment]
@doxygen Yes, problem solved.

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.0.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
